### ISSUE TITLE
Feature Request: Notebook to script

### ISSUE THREAD

**Issue by betteridiot** [jupyterlab/jupyterlab#3732]
Small quality of life feature request. We can already drag & drop from one notebook to another notebook. However, it would be awesome if you could drag a cell from a notebook into the text editor and have it populate that cell's script into the editor. Don't know if this is possible, but I like to think of my notebook as a checklist of sorts. When a function for a module is done being prototyped, I like getting it out of my sight so to speak.

**Comment by carlthome** [comment]
I like this, but if you only do this for a single cell the created module will typically not work as there are globals in the notebook (set by previous cell runs).

There should preferably be some smartness added to either warn if the cell relies on other cells somehow, and maybe required imports, top-level objects, and the like could even be automatically added to the export.

**Comment by holdenweb** [comment]
I imagine a naive implementation that simply treated the cell contents as text would be good enough for 80% of purposes, and it would certainly avoid a LOT of complex analysis that would be required to add the "smartness," which could be provided by a later enhancement if necessary. Cross-process drag and drop seems like a pretty complicated and platform-dependent feature, TBH.

**Comment by ian-r-rose** [comment]
I agree with @holdenweb that actually populating the session with imports, variables, etc would be prohibitively complex (especially since we support order 100 kernels). That being said, I think that drag-drop for cell contents into a text editor would be very doable (perhaps for an advanced beginner).

**Comment by spapini** [comment]
What about https://github.com/jupyter/nbconvert ?
It support converting notebooks to scripts.
We can probably use it to add something like "export to script"

**Comment by holdenweb** [comment]
What would that do that "File | Download as | Python" wouldn't?

**Comment by spapini** [comment]
Wait, it already exists, you are right. My bad. (File -> export as -> executable script)
